code/entities/cooking/hotdog-roller/interactable/Roller.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Conna.Inventory;
+using Sandbox;
+using Sandbox.util;
+using System.Collections.Generic;
+
+namespace Cinema.HotdogRoller;
+
+public class Roller : BaseInteractable
+{
+    private struct Slot
+    {
+        public HotdogCookable Entity { get; set; }
+        public string Attachment { get; set; }
+
+        public Slot(string attachment)
+        {
+            Attachment = attachment;
+        }
+    }
+
+    private struct SlotDist
+    {
+        public Slot Slot { get; set; }
+        public float Distance { get; set; }
+        public int Index { get; set; }
+        public Transform SlotTransform { get; set; }
+
+        public SlotDist(Slot slot, float dist, int index, Transform t)
+        {
+            Slot = slot;
+            Distance = dist;
+            Index = index;
+            SlotTransform = t;
+        }
+    }
+
+    public Knob Knob { get; set; }
+    public Switch Switch { get; set; }
+    private bool front;
+    private Slot[] slots = new Slot[10];
+
+    public Roller() // For the compiler...
+    {
+    }
+
+    public Roller(Switch p_switch, Knob knob, bool front = false)
+    {
+        Switch = p_switch;
+        Knob = knob;
+        this.front = front;
+
+        for (int i = 0; i < 10; i++)
+        {
+            slots[i] = new Slot($"S{i + 1}{(front ? 'F' : 'B')}");
+        }
+    }
+
+    private HotdogCookable addHotdog(Roller parent, Transform t)
+    {
+        var hotdog = new HotdogCookable(parent);
+        var reverse = Game.Random.Int(1) == 1;
+
+        hotdog.Transform = t;
+        hotdog.LocalRotation = hotdog.LocalRotation.RotateAroundAxis(Vector3.Forward, Game.Random.Float(180));
+        hotdog.LocalRotation = hotdog.LocalRotation.RotateAroundAxis(Vector3.Up, reverse ? 180 : 0);
+        hotdog.Parent = Parent;
+
+        return hotdog;
+    }
+
+    /// <summary>
+    /// This will add or take the hotdog you are pressing closest to, if nothing is close it will just add a hotdog to the nearest possible slot.
+    /// </summary>
+    /// <param name="ply"></param>
+    public override void Trigger(Player ply)
+    {
+        List<SlotDist> slotsIndexed = new List<SlotDist>();
+        var parent = Parent as HotdogRoller;
+
+        for (int i = 0; i < 10; i++)
+        {
+            var slotData = slots[i];
+
+            if (parent.GetAttachment(slotData.Attachment) is Transform t)
+            {
+                var dist = LastTriggerResults.HitPosition.Distance(t.Position);
+
+                slotsIndexed.Add(new SlotDist(slotData, dist, i, t));
+            }
+        }
+
+        slotsIndexed.Sort((x, y) => { return x.Distance.CompareTo(y.Distance); });
+
+        foreach(var slotCompared in slotsIndexed)
+        {
+            var slotData = slotCompared.Slot;
```

Comment:
"data" should be avoided in variable names. Everything is data. By calling something XData you aren't really providing any more context to another developer.

Consider naming this something else. Or if you're struggling because your struct has a member of a `Slot` and you're calling the instance of that struct a "slot something" then maybe you've misnamed the struct.

"Why would a slot have a slot inside of it?"